projects/packages/connection/src/class-heartbeat.php

```diff
@@ -249,4 +256,55 @@ public function cli_callback( $args ) {
 			WP_CLI::line( sprintf( __( 'Last heartbeat sent at: %s', 'jetpack-connection' ), $last_date ) );
 		}
 	}
+
+	/**
+	 * Initialize the heartbeat REST API.
+	 *
+	 * @return void
+	 */
+	public function initialize_rest_api() {
+		register_rest_route(
+			'jetpack/v4',
+			'/get_heartbeat_data',
```

Comment:
I haven't reviewed the entire thing yet, but do you think that `get_heartbeat_data` is a redundant name? I don't think snake case is ever used in REST API, and even if it was, we are already specifying that it is a `GET` with an HTTP verb. And `data` is like - what else can you get from an API endpoint? :)

Reply:
`/jetpack/v4/heartbeat` looks a lot better to me

Reply:
Good idea.
I named it `/jetpack/v4/heartbeat/data` to emphasize there's data being retrieved 🙂